java/core/src/main/java/co/worklytics/psoxy/storage/impl/CSVFileHandler.java

```diff
@@ -0,0 +1,100 @@
+package co.worklytics.psoxy.storage.impl;
+
+import co.worklytics.psoxy.storage.FileHandler;
+import co.worklytics.psoxy.Rules;
+import co.worklytics.psoxy.Sanitizer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.inject.Inject;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@NoArgsConstructor(onConstructor_ = @Inject)
+public class CSVFileHandler implements FileHandler {
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Override
+    public byte[] handle(@NonNull InputStreamReader reader, @NonNull Sanitizer sanitizer) throws IOException {
+        CSVParser records = CSVFormat.DEFAULT
+                .withFirstRecordAsHeader()
+                .parse(reader);
+
+        Preconditions.checkArgument(records.getHeaderMap() != null, "Failed to parse header from file");
+
+        Map<Integer, String> invertedHeaderMap = records.getHeaderMap().entrySet().stream()
```

Comment:
I don't think this is necessary, just have the values around so using `records.getHeaderMap().keySet()` should be enough